browser/ui/tabs/brave_tab_menu_model.cc

```diff
@@ -111,4 +119,57 @@ void BraveTabMenuModel::Build(int selected_tab_count) {
   InsertItemWithStringIdAt(close_other_tabs_index.value_or(GetItemCount()),
                            CommandCloseDuplicateTabs,
                            IDS_TAB_CXMENU_CLOSE_DUPLICATE_TABS);
+
+  if (base::FeatureList::IsEnabled(tabs::features::kBraveSplitView)) {
+    BuildItemsForSplitView(browser, tab_strip_model, indices);
+    return;
+  }
+}
+
+void BraveTabMenuModel::BuildItemsForSplitView(
+    Browser* browser,
+    TabStripModel* tab_strip_model,
+    const std::vector<int>& indices) {
+  CHECK(browser);
+
+  auto index = *GetIndexOfCommandId(TabStripModel::CommandReload);
+
+  // In case only one tab is selected
+  //  * if the tab is tiled, show "Close Split View" and "Break into Tabs"
+  //  * else show "New Split View"
+  auto* split_view_data = SplitViewBrowserData::FromBrowser(browser);
+  if (indices.size() == 1u) {
+    auto tab_handle = tab_strip_model->GetTabHandleAt(indices.front());
+    if (split_view_data->IsTabTiled(tab_handle)) {
+      InsertItemWithStringIdAt(++index, CommandCloseSplitView,
+                               IDS_IDC_CLOSE_SPLIT_VIEW);
+      InsertItemWithStringIdAt(++index, CommandBreakTile,
+                               IDS_TAB_CXMENU_BREAK_TILE);
+      return;
+    }
+
+    InsertItemWithStringIdAt(++index, CommandNewSplitView,
+                             IDS_IDC_NEW_SPLIT_VIEW);
+    return;
+  }
+
+  // In case two tabs are selected:
+  if (indices.size() == 2) {
+    auto tab_handle1 = tab_strip_model->GetTabHandleAt(indices[0]);
+    auto tab_handle2 = tab_strip_model->GetTabHandleAt(indices[1]);
+    if (split_view_data->GetTile(tab_handle1) ==
+        split_view_data->GetTile(tab_handle2)) {
+      InsertItemWithStringIdAt(++index, CommandTileTabs,
```

Comment:
Note: I can select two tiled tabs and attempt to `TILE_TABS` which results in a crash. Maybe we shouldn't show this command if any of the selected tabs is currently tiled?

Reply:
Well spotted, we shouldn't mix up two tiled tabs.